Source/Core/Request.js

```diff
@@ -0,0 +1,162 @@
+     * @param {Boolean} [options.url] The url to request.
+     * @param {Function} [options.requestFunction] The actual function that makes the request. The function takes no arguments and returns a promise for the requested data.
+     * @param {Boolean} [options.throttle=false] Whether to throttle and prioritize the request. If false, the request will be sent immediately. If true, the request will be throttled and sent based on priority.
+     * @param {Boolean} [options.throttleByServer=false] Whether to throttle the request by server.
+     * @param {RequestType} [options.type=RequestType.OTHER] The type of request.
+     * @param {Number} [options.distance=0.0] The distance from the camera, used to prioritize requests.
+     * @param {Number} [options.screenSpaceError=0.0] The screen space error, used to prioritize requests.
+     */
+    function Request(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var throttleByServer = defaultValue(options.throttleByServer, false);
+        var throttle = throttleByServer || defaultValue(options.throttle, false);
+
+        /**
+         * The URL to request.
+         *
+         * @type {String}
+         */
+        this.url = options.url;
+
+        /**
+         * The actual function that makes the request. The function takes no arguments and returns a promise for the requested data.
+         *
+         * @type {Function}
+         *
+         * @private
+         */
+        this.requestFunction = options.requestFunction;
+
+        /**
+         * Whether to throttle and prioritize the request. If false, the request will be sent immediately. If true, the
+         * request will be throttled and sent based on priority.
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        this.throttle = throttle;
+
+        /**
+         * Whether to throttle the request by server. Browsers typically support about 6-8 parallel connections
+         * for HTTP/1 servers, and an unlimited amount of connections for HTTP/2 servers. Setting this value
+         * to <code>true</code> is preferable for requests going through HTTP/1 servers.
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        this.throttleByServer = throttleByServer;
+
+        /**
+         * Type of request.
+         *
+         * @type {RequestType}
+         * @readonly
+         *
+         * @default RequestType.OTHER
+         */
+        this.type = defaultValue(options.type, RequestType.OTHER);
+
+        /**
+         * The distance from the camera, used to prioritize requests. This value may be edited continually to update
+         * the request's priority.
+         *
+         * @type {Number}
+         *
+         * @default 0.0
+         */
+        this.distance = defaultValue(options.distance, 0.0);
```

Comment:
This is a leaky abstraction.  It's odd to have `distance` and `screenSpaceError` on the Request object.  I understand that there needs to be a metric to prioritize request, but why not just have this be a function that returns a unitless priority value and then standard implementations for world-based objects?

I can also imagine other odd situations where you are loading 100 instances of the same model.  Who's position do you use for the distance calculation in the queue?  My turning this into a callback, each RequestScheduler user can optimize these calls for their needs.